src/download.ts

```diff
@@ -75,23 +75,24 @@ export class Download {
   }
 
   async saveAs(path: string) {
-    if (this._loaded) {
+    if (!this._acceptDownloads)
+      throw new Error('Pass { acceptDownloads: true } when you are creating your browser context.');
+    if (this._deleted)
+      throw new Error('Download already deleted. Save before deleting.');
+    if (this._failure)
+      throw new Error('Download not found on disk. Check download.failure() for details.');
+
+    if (this._finished) {
```

Comment:
I would just `await this._finishedPromise` here and continue with saving as. That way we would not have to save from the `_reportFinished` event handler and jump around with fulfill/reject.

Reply:
This whole point of this dance is to make sure we save before we notify listeners...